mlflow/models/evaluation/default_evaluator.py

```diff
@@ -494,28 +522,51 @@ def _log_model_explainability(self):
             f: f2 for f, f2 in zip(self.feature_names, truncated_feature_names)
         }
 
-        sampled_X = shap.sample(self.X, sample_rows)
+        sampled_X = shap.sample(self.X, sample_rows, random_state=0)
+        if algorithm == "kernel":
+            background_X = shap.sample(self.X, sample_rows, random_state=3)
+        else:
+            background_X = None
 
         if isinstance(sampled_X, pd.DataFrame):
             # For some shap explainer, the plot will use the DataFrame column names instead of
             # using feature_names argument value. So rename the dataframe column names.
             sampled_X = sampled_X.rename(columns=truncated_feature_name_map, copy=False)
+            if background_X is not None:
+                background_X = background_X.rename(columns=truncated_feature_name_map, copy=False)
+        else:
+            # If sample_X is numpy array, convert to pandas dataframe.
+            sampled_X = pd.DataFrame(sampled_X, columns=truncated_feature_name_map)
```

Comment:
`truncated_feature_name_map` is a dict, right? What does `sampled_X.columns` look like if `truncated_feature_name_map` is `{"a": "b", "c": "d"}`?

Reply:
Quick test:


```python
>>> pd.DataFrame([[1,2], [3,4]], columns={"a": "b", "c": "d"}).columns
Index(['a', 'c'], dtype='object')
```

Is this the desired result?

Reply:
My fault. Fix it. Should use `truncated_feature_names` here.